modules/networking/default.nix

```diff
@@ -0,0 +1,41 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.networking;
+
+  hostName = optionalString (cfg.hostName != null) ''
+    scutil --set ComputerName "${cfg.hostName}"
+    sudo scutil --set LocalHostName "${cfg.hostName}"
```

Comment:
No need for sudo, the `system.activationScripts` run as root. If the first command needs to run as the current user it should be included in the user activation.